src/defaults/defaults_linux.go

```diff
@@ -7,7 +7,7 @@ package defaults
 func GetDefaults() platformDefaultParameters {
 	return platformDefaultParameters{
 		// Admin
-		DefaultAdminListen: "unix:///var/run/yggdrasil.sock",
+		DefaultAdminListen: "unix:///var/run/yggdrasil/yggdrasil.sock",
```

Comment:
This breaks running if the `/var/run/yggdrasil` folder doesn't exist, so the default depends on the systemd package (or otherwise creating the folder). The binary produced by `./build` needs to be able to run with default settings (i.e. `-autoconf`) with no further changes to the OS/environment.

Reply:
I didn't test this and indeed if yggdrasil does not try to create the directory in which it wants to place the socket (which IMOHO is a bug) then this will fail.

An actual go progammer is required to make ygg create the directory before it attempts to create a file in it (assuming this doesn't work as you state). I am not a Go programmer, would appreciate a patch. I guess its not too complex.

Reply:
No problem, default socket path can be different if run with `-autoconf`. Probably, it should be created in user's home directory if run this way.

Reply:
good point. Likely best to use /run/user/{UID}/ which is where Linux by default makes those.

For instance wayland creates its sockets there.

Reply:
No, it's bad to hardcode this path. The better way would be to use contents `XDG_RUNTIME_DIR` environment variable if it's present, otherwise something like `.yggdrasil.sock` in user's home directory. I presume, this will be a good default for those who still insist on not using `systemd`.

Reply:
Actually, defaults should be dependent on whether config is generated by regular user or `root` (not on `-autoconf` option).

Regular user should have everything in the home directory (and socket `$XDG_RUNTIME_DIR` if this variable is present).
But for `root` user everything should be in LSB-conforming directories (config in `/etc/yggdrasil`, socket in `/run/yggdrasil`).